austation/code/modules/food_and_drinks/food/kitchen_machinery/cake_printer.dm

```diff
@@ -0,0 +1,177 @@
+		to_chat(user, "<span class='warning'>Close the maintenance panel first.</span>")
+		return ..()
+	if(istype(I, /obj/item/reagent_containers))
+		if(I.reagents.has_reagent(/datum/reagent/consumable/synthetic_cake_batter))
+			if(fuel >= max_fuel)
+				to_chat(user, "<span class='warning'>The machine's tank is full!</span>")
+				return ..()
+			fuel += (10 * (I.reagents.get_reagent_amount(/datum/reagent/consumable/synthetic_cake_batter)))
+			to_chat(user, "<span class='notice'>You pour the cake batter in [src].</span>")
+			I.reagents.remove_reagent(/datum/reagent/consumable/synthetic_cake_batter, I.reagents.get_reagent_amount(/datum/reagent/consumable/synthetic_cake_batter))
+			if(fuel > max_fuel)
+				fuel = max_fuel
+			return ..()
+	if(cake_blacklist[I.type] || HAS_TRAIT(I, TRAIT_NODROP) || (I.item_flags & (ABSTRACT | DROPDEL)))
+		return ..()
+	item_scanned = I
+	switch(item_scanned.w_class)
+		if(WEIGHT_CLASS_TINY)
+			fuelcost = 15 / efficiency
+		if(WEIGHT_CLASS_SMALL)
+			fuelcost = 30 / efficiency
+		if(WEIGHT_CLASS_NORMAL)
+			fuelcost = 50 / efficiency
+		if(WEIGHT_CLASS_HUGE)
+			fuelcost = 100 / efficiency
+		if(WEIGHT_CLASS_GIGANTIC)
+			fuelcost = 200 / efficiency
+	to_chat(user, "<span class='notice'>You scan [item_scanned] on [src].</span>")
+
+/obj/machinery/cake_printer/examine(mob/user)
+	. = ..()
+	if(in_range(user, src) || isobserver(user))
+		. += "<span class='notice'>It has <b>[fuel]</b> fuel left.</span>"
+		if(item_scanned)
+			. += "<span class='notice'>Alt-click to reset the scanner, current scanned item is [item_scanned].</span>"
+		else
+			. += "<span class='notice'>Alt-click to reset the scanner, no items are scanned.</span>"
+		. += "<span class='notice'>The status display reads: Fuel consumption reduced by <b>[(efficiency*25)-25]</b>%.<br>Machine can hold up to <b>[max_fuel]</b> units of fuel.<br> Speed is increased by <b>[(speed*100)-100]%</b></span>"
+		if(obj_flags & EMAGGED)
+			. += "<span class='warning'>It's status display looks a bit burnt!</span>"
+		. += "<span class='notice'>The machine takes synthetic cake batter as fuel, which is some flour, some milk and a bit of astrotame.</span>"
+
+/obj/machinery/cake_printer/Destroy()
+	if(caked_item)
+		caked_item.forceMove(drop_location())
+		caked_item = null
+	item_scanned = null
+	. = ..()
+
+/obj/machinery/cake_printer/attack_ai(mob/user)
+	return
+
+/obj/machinery/cake_printer/attack_hand(mob/user)
+	if(processing)
+		to_chat(user, "<span class='warning'>\The [src] is printing!</span>")
+		return
+	if(panel_open)
+		to_chat(user, "<span class='warning'>Close the maintenance panel first!</span>")
+		return
+	if(fuel <= 0)
+		to_chat(user, "<span class='warning'>No cake batter in the printer!</span>")
+		return
+	if(fuelcost > fuel)
+		to_chat(user, "<span class='warning'>Not enough fuel, add more cake batter!</span>")
+		return
+	if(!item_scanned)
+		to_chat(user, "<span class='warning'>Scan an item first!</span>")
+		return
+	if(!processing)
```

Comment:
processing is already checked above

Reply:
it's in that order for a reason

Reply:
Doing double checks can almost always be avoided

Reply:
not in this condition